### ISSUE TITLE
Possible to reorder multiple attached files without reuploading?

### ISSUE THREAD

**Issue by veborich** [carrierwaveuploader/carrierwave#2337]
Hi say I want a `user` to be able to upload multiple `avatars`, but I want them to be able to set their default avatar. Since the only record I can see of `avatars` in the db is the array of filenames, I figured the best way to set the "default" is have it be at the zero index.

Using [this guide](https://github.com/carrierwaveuploader/carrierwave/wiki/How-to:-Add-more-files-and-remove-single-file-when-using-default-multiple-file-uploads-feature) on modifying instead of replacing uploads.

The guide shows example code of making an array copy, changing the array, then reassign it to the model's uploads. I have done this as below:

```
    index = params[:index].to_i
    all_avatars = @user.avatars
    old_default_avatar = all_avatars[0]
    new_default_avatar = all_avatars[index]
    all_avatars[0] = new_default_avatar
    all_avatars[index] = old_default_avatar
    @user.avatars = all_avatars
```

And it works!  But it works by reuploading ALL the avatars all over again.  This makes the action itself pretty slow, and is going to be much worse for larger images that aren't avatars.  Am I missing something obvious?  Is there perhaps a way to add a validation that if the upload hasn't changed position, don't reupload?

**Comment by BryanH** [comment]
Ask support questions on [StackOverflow](https://stackoverflow.com/questions/tagged/carrierwave)

**Comment by mshibuya** [comment]
Closed by #2401.
